code/game/machinery/status_display.dm

```diff
@@ -148,38 +194,133 @@
 		(current_mode == SD_BLANK) || \
 		(current_mode != SD_PICTURE && message1 == "" && message2 == "") \
 	)
-		set_light(0)
+		set_light(l_on = FALSE)
 		return
-	set_light(1.4, 0.7, LIGHT_COLOR_BLUE) // blue light
+
+	set_light(l_color = text_color, l_on = TRUE)
 
 /obj/machinery/status_display/update_overlays()
 	. = ..()
 
+	// Facing south, we render traditionally.
+	if(dir == SOUTH)
+		add_screen_overlays(.)
+		return
+
+	// Otherwise, we render a projection on the floor.
+
+	// Get screen overlays and return if it's off.
+	var/list/projected_overlays = list()
+	var/anything_displayed = add_screen_overlays(projected_overlays, projection_only = TRUE)
+
+	if(!anything_displayed)
+		return
+
+	// Both of these are vis contents that will receive 1 or more overlays themselves.
+
+	// Matrices to transform the content given the direction
+	var/matrix/floor_matrix = floor_projections["[dir]"]
+	var/matrix/beam_matrix = beam_projections["[dir]"]
+
+	// Make the 2 vis overlays.
+	generate_projection_overlay(floor_matrix, projected_overlays, alpha = PROJECTION_FLOOR_ALPHA)
+	generate_projection_overlay(beam_matrix, projected_overlays, alpha = PROJECTION_BEAM_ALPHA)
+
+	// Impossible for this to look good as an actual emissive since the emissive plane crushes alpha to all or nothing for some reason.
```

Comment:
for the record this is done because of annoying requirements for filtering emissives

Reply:
I've removed the "for some reason" part.